packages/react-components/react-tree/src/utils/createUnfilteredFlatTree.ts

```diff
@@ -0,0 +1,81 @@
+import { FlatTreeItem, FlatTreeItemProps } from '../hooks/useFlatTree';
+
+/**
+ * @internal
+ */
+export type UnfilteredFlatTree = {
+  items: FlatTreeItem[];
+  itemsPerId: Map<string, FlatTreeItem>;
+};
+
+/**
+ * creates a list of flat tree items
+ * and provides a map to access each item by id
+ */
+export function createUnfilteredFlatTree(flatTreeItemProps: FlatTreeItemProps[]): UnfilteredFlatTree {
+  const root = createRootFlatTreeItem();
+  const itemsPerId = new Map<string, FlatTreeItem>([[root.id, root]]);
+  const items: FlatTreeItem[] = [];
+
+  for (let index = 0; index < flatTreeItemProps.length; index++) {
+    const { parentId = flatTreeRootParentId, ...treeItemProps } = flatTreeItemProps[index];
+
+    const nextItemProps = flatTreeItemProps[index + 1] as FlatTreeItemProps | undefined;
+    const currentParent = itemsPerId.get(parentId);
+    if (!currentParent) {
+      if (process.env.NODE_ENV === 'development') {
+        // eslint-disable-next-line no-console
+        console.error(
+          `useFlatTree: item ${flatTreeItemProps[index].id} not properly initialized, make sure provided items are organized`,
+        );
+      }
+      break;
+    }
+    const isLeaf = nextItemProps?.parentId !== treeItemProps.id;
+    const currentLevel = (currentParent.level ?? 0) + 1;
+    const currentChildrenSize = ++currentParent.childrenSize;
+
+    const flatTreeItem: FlatTreeItem = {
+      id: treeItemProps.id,
+      getTreeItemProps: () => ({
+        ...treeItemProps,
+        'aria-level': currentLevel,
+        'aria-posinset': currentChildrenSize,
+        'aria-setsize': currentParent.childrenSize,
+        leaf: isLeaf,
+      }),
+      level: currentLevel,
+      parentId,
+      childrenSize: 0,
+      index: -1,
+    };
+    itemsPerId.set(flatTreeItem.id, flatTreeItem);
```

Comment:
each key is just a simple string, is there a need to use an ES6 map in this case?

Reply:
Honestly, since it's basically an immutable map that never changes after it's creation.... no there is no need, it was just a preference